function.js

```diff
@@ -1,77 +1,43 @@
-  document.getElementById("misc").style.visibility = "hidden";
-}
-
-function myWork() {
-  document.getElementById("home").style.visibility = "hidden";
-  document.getElementById("opening").style.visibility = "hidden";
-  document.getElementById("skillset").style.visibility = "hidden";
-  document.getElementById("work").style.visibility = "visible";
-  document.getElementById("education").style.visibility = "hidden";
-  document.getElementById("interest").style.visibility = "hidden";
-  document.getElementById("reason").style.visibility = "hidden";
-  document.getElementById("misc").style.visibility = "hidden";
-}
-
-function myStudy() {
-  document.getElementById("home").style.visibility = "hidden";
-  document.getElementById("opening").style.visibility = "hidden";
-  document.getElementById("skillset").style.visibility = "hidden";
-  document.getElementById("work").style.visibility = "hidden";
-  document.getElementById("education").style.visibility = "visible";
-  document.getElementById("interest").style.visibility = "hidden";
-  document.getElementById("reason").style.visibility = "hidden";
-  document.getElementById("misc").style.visibility = "hidden";
-}
-
-function myHobby() {
-  document.getElementById("home").style.visibility = "hidden";
-  document.getElementById("opening").style.visibility = "hidden";
-  document.getElementById("skillset").style.visibility = "hidden";
-  document.getElementById("work").style.visibility = "hidden";
-  document.getElementById("education").style.visibility = "hidden";
-  document.getElementById("interest").style.visibility = "visible";
-  document.getElementById("reason").style.visibility = "hidden";
-  document.getElementById("misc").style.visibility = "hidden";
-}
-
-function myReason() {
-  document.getElementById("home").style.visibility = "hidden";
-  document.getElementById("opening").style.visibility = "hidden";
-  document.getElementById("skillset").style.visibility = "hidden";
-  document.getElementById("work").style.visibility = "hidden";
-  document.getElementById("education").style.visibility = "hidden";
-  document.getElementById("interest").style.visibility = "hidden";
-  document.getElementById("reason").style.visibility = "visible";
-  document.getElementById("misc").style.visibility = "hidden";
-}
-
-function myOthers() {
-  document.getElementById("home").style.visibility = "hidden";
-  document.getElementById("opening").style.visibility = "hidden";
-  document.getElementById("skillset").style.visibility = "hidden";
-  document.getElementById("work").style.visibility = "hidden";
-  document.getElementById("education").style.visibility = "hidden";
-  document.getElementById("interest").style.visibility = "hidden";
-  document.getElementById("reason").style.visibility = "hidden";
-  document.getElementById("misc").style.visibility = "visible";
-}
-
+function scrollingTop() {
+  const y = window.scrollY;
+  if(y > 0){
+    window.requestAnimationFrame(scrollingTop);
```

Comment:
If you want to write a recursive function, I think it's clearer to define the exit condition clearly rather than inside this if condition with additional side effects after the call.